sharkfuser/include/fusilli/logging.h

```diff
@@ -337,4 +342,36 @@ inline ConditionalStreamer &getLogger() {
     }                                                                          \
   } while (false);
 
+// Unwrap the type returned from an expression that evaluates to an ErrorOr,
+// returning an error from the enclosing function in the error case, and the
+// value otherwise.
+//
+// Usage:
+//   ErrorOr<std::string> getString();
+//
+//   ErrorOr<int> processString() {
+//     // Either gets the string or returns error.
+//     std::string str = FUSILLI_TRY(getString());
+//     return ok(str.length());
+//   }
+#define FUSILLI_TRY(expr)                                                      \
+  ({                                                                           \
+    auto _errorOr = (expr);                                                    \
+    if (isError(_errorOr)) {                                                   \
+      FUSILLI_LOG_LABEL_RED("ERROR: ");                                        \
+      FUSILLI_LOG_ENDL(#expr << " at " << __FILE__ << ":" << __LINE__);        \
+      return ErrorObject(_errorOr);                                            \
+    }                                                                          \
+    std::move(*_errorOr);                                                      \
+  })
+
+// Test version of FUSILLI_TRY that uses Catch2 REQUIREs instead of returning an
+// ErrorObject.
+#define TEST_FUSILLI_TRY(expr)                                                 \
+  ({                                                                           \
+    auto _errorOr = (expr);                                                    \
+    REQUIRE(isOk(_errorOr));                                                   \
+    std::move(*_errorOr);                                                      \
+  })
```

Comment:
Do we really need this? My worry is divergence between the two over time. Any issue with using FUSILLI_TRY and check for isOk or isError at the callsite (tests) itself?

Reply:
> The intent of FUSILLI_TRY is to propagate an error back to the caller in the case where a callee can't handle it, the intent of TEST_FUSILLI_TRY is to deduplicate an often repeated block of code in tests.

We don't want to take a dep on `Catch2` for our main fusilli lib (this is an optional dep if users build it with tests skipped). Why not just define this macro as a test helper (in the testing script itself, or a common helper that lives in tests if this is used by >1 tests).

Reply:
Sounds good 👍